R/prepare_GISTIC_inputs.R

```diff
@@ -0,0 +1,234 @@
+#' @param projection
+#' @param output_dir
+#' @param flavour
+#' @param overwrite
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#'
+#' BL_meta = get_gambl_metadata() %>% dplyr::filter(pathology=="BL")
+#' prepare_gistic_inputs(BL_meta,"hg38","/Users/rmorin/Desktop/GISTIC2/BL/","battenberg")
+#'
+#'
+#' FL_meta = get_gambl_metadata() %>% dplyr::filter(pathology=="FL",seq_type!="mrna")
+#' prepare_gistic_inputs(FL_meta,"grch37","/Users/rmorin/Desktop/GISTIC2/FL/","battenberg")
+prepare_gistic_inputs <- function(these_samples_metadata,
+                                  projection = "grch37",
+                                  output_dir = "/Users/rmorin/Desktop/GISTIC2/",
+                                  flavour = "combined",
+                                  overwrite = FALSE) {
+    #check that output directory exists
+    if(!dir.exists(output_dir)){
+      stop("you must provide an existing path")
+    }
+    markers <- paste0(output_dir, "markers")
+    out_seg <- paste0(output_dir, "all.seg")
+    if(!overwrite){
+      if(file.exists(markers)|file.exists(out_seg)){
+        print("output files already exist in this directory")
+        stop("If you want to clobber these, re-run with overwrite = TRUE")
+      }
+    }
+    # get the seg_data
+    seg_for_gistic <- get_cn_segments(
+        these_samples_metadata = these_samples_metadata,
+        flavour = flavour,
+        fill_missing_with = "avg_ploidy",
+        projection = projection
+    )
+    #check how many samples we have data from
+    u_id = length(unique(seg_for_gistic$ID))
+    print(paste("got data from",u_id,"sample IDs"))
+    seg_for_gistic <- seg_for_gistic %>%
+        dplyr::select(ID:log.ratio) %>%
+        mutate(LOH_flag = ifelse(LOH_flag == 99, 0, LOH_flag))
+    print("checking for overlapping segments")
+    seg_for_gistic <- check_overlap(seg_for_gistic)
+
+    if ("overlap" %in% seg_for_gistic$overlap_status) {
+        print("resolving overlapping segments")
+        seg_for_gistic <- solve_overlap(seg_for_gistic)
+    } else {
+      seg_for_gistic <- seg_for_gistic %>%
+            select(-overlap_status, -region_size)
+    }
+
+    # write to disk and make markers file
+    write_tsv(seg_for_gistic, file = out_seg)
+    temp_markers <- paste0(output_dir, "temp_markers")
+    print("processing for GISTIC")
+    cmd <- paste("sed '1d'", out_seg, "| cut -f2,3 > ", temp_markers)
+    system(cmd)
+    cmd <- paste("sed '1d'", out_seg, "| cut -f2,4 >> ", temp_markers)
+    system(cmd)
+    cmd <- paste("sort -V -k1,1 -k2,2nr", temp_markers, "|uniq | nl > ", markers)
+    system(cmd)
+    # sed '1d' {input.seg} | cut -f2,3 > {output.temp_markers}
+    # sed '1d' {input.seg} | cut -f2,4 >> {output.temp_markers}
+    # sort -V -k1,1 -k2,2nr {output.temp_markers} | uniq | nl > {output.markers}
+    print("done!")
+    print("Running gistic:")
+    print("conda activate /projects/rmorin_scratch/conda_environments/30ee048690c736b4964ed302c88be45f_")
```

Comment:
should we remove these commands? They refer to GSC path which will not exist for external users, and to the input and reference files that are not necessarily generated or provided with this function (the ./inputs may or may not exist and refgene.mat is not provided + the seg file may be in different projection then what the print statement refers to). Alternatively, we can place this in the .results and modify the commands so the path/files can be found